File: plugin/intellij/src/main/kotlin/com/github/jshwi/docsig/ui/ClassCheckMode.kt
/**
 * Enumeration for the class check mode.
 */
package com.github.jshwi.docsig.ui

/**
 * Maps class-check UI choices to optional CLI flags.
 *
 * @property label Combo box caption shown to the user.
 * @property flag CLI flag when present, or null when omitted.
 */
internal enum class ClassCheckMode(val label: String, val flag: String?) {
    NONE(
        "settings.options.class-checking-mode-none",
        null,
    ),
    CLASS(
        "settings.options.class-checking-mode-check-class",
        "--check-class",
    ),
    CLASS_CONSTRUCTOR(
        "settings.options.class-checking-mode-check-class-constructor",
        "--check-class-constructor",
    ),
}

File: plugin/vscode/src/settings/ClassCheckMode.ts
export enum ClassCheckMode {
  None = "none",
  CheckClass = "checkClass",
  CheckClassConstructor = "checkClassConstructor",
}

const FLAGS: Record<ClassCheckMode, string | null> = {
  [ClassCheckMode.None]: null,
  [ClassCheckMode.CheckClass]: "--check-class",
  [ClassCheckMode.CheckClassConstructor]: "--check-class-constructor",
};

export function classCheckFlag(mode: ClassCheckMode): string | null {
  return FLAGS[mode];
}
